<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
<xacro:include filename="inertial_macros.xacro" />

<!--materials-->

<material name="Blue">
    <color rgba="0 0 0.7 1" />
</material>
<material name="Green">
    <color rgba="0 0.7 0.2 1" />
</material>
<material name="Red">
    <color rgba="0.7 0.1 0.1 1" />
</material>
<material name="Grey">
    <color rgba="0.5 0.5 0.5 1" />
</material>
<material name="White">
    <color rgba="0.9 0.8 0.8 1" />
</material>
<material name="Black">
    <color rgba="0 0 0 1" />
</material>

<!--link and joints-->
<!--        pour ce projet base_link = chassis       -->

<link name="base_footprint"></link>
<joint name="base_footprin_joint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link"/>
    <origin xyz="0 0 0.1425" rpy="0 0 0" />
</joint>

<!--base_link-->

<link name="base_link">
    <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
            <box size="0.1 0.17 0.22" />
        </geometry>
        <material name="White">
        </material>
    </visual>
    <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
            <box size="0.1 0.17 0.22" />
        </geometry>
    </collision>

    <xacro:inertial_box mass="1.0" x="0.1" y="0.17" z="0.22" >
        <origin xyz="0 0 0" rpy="0 0 0" />
    </xacro:inertial_box>
</link>
<gazebo reference="base_link">
  <material>Gazebo/Red</material>
</gazebo>

<!--            wheels           -->

<!--Right_Wheel_link-->
<link name="RightWheel_link">
    <visual>
        <origin xyz="0 0 0" rpy="1.57 0 0" />
        <geometry>
            <cylinder radius="0.07" length="0.025" />
        </geometry>
        <material name="Blue">
        </material>
    </visual>
    <collision>
        <origin xyz="0 0 0" rpy="1.57 0 0" />
        <geometry>
            <cylinder radius="0.07" length="0.025" />
        </geometry>
    </collision>

    <xacro:inertial_cylinder mass="0.5" radius="0.07" length="0.025" >
        <origin xyz="0 0 0" rpy="0 0 0" />
    </xacro:inertial_cylinder>
</link>
<gazebo reference="RightWheel_link">
  <material>Gazebo/Green</material>
  <mu1>1.0</mu1>
  <mu2>1.0</mu2>
</gazebo>

<!--Right_Wheel_joint-->
<joint name="base_Rwheel_joint" type="continuous">
    <parent link="base_link" />
    <child link="RightWheel_link" />
    <origin xyz="0 -0.0975 -0.11" rpy="0 0 0" />
    <axis xyz="0 1 0" />
    <dynamics damping="0.5" friction="0.01"/> <!--hadi friction t3 joint ( tsma moteur mchi les roues )-->
    <limit effort="50.0" velocity="20.0"/>
</joint>

<!--Left_Wheel_link-->
<link name="LeftWheel_link">
    <visual>
        <origin xyz="0 0 0" rpy="1.57 0 0" />
        <geometry>
            <cylinder radius="0.07" length="0.025" />
        </geometry>
        <material name="Blue">
        </material>
    </visual>
    <collision>
        <origin xyz="0 0 0" rpy="1.57 0 0" />
        <geometry>
            <cylinder radius="0.07" length="0.025" />
        </geometry>
    </collision>

    <xacro:inertial_cylinder mass="0.5" radius="0.07" length="0.025" >
        <origin xyz="0 0 0" rpy="0 0 0" />
    </xacro:inertial_cylinder>
</link>
<gazebo reference="LeftWheel_link">
  <material>Gazebo/Green</material>
  <mu1>1.0</mu1>
  <mu2>1.0</mu2>
</gazebo>

<!--Right_Wheel_joint-->
<joint name="base_Lwheel_joint" type="continuous">
    <parent link="base_link" />
    <child link="LeftWheel_link" />
    <origin xyz="0 0.0975 -0.11" rpy="0 0 0" />
    <axis xyz="0 1 0" />
    <dynamics damping="0.5" friction="0.01"/> 
    <limit effort="50.0" velocity="20.0"/>
</joint>





</robot>